diff --git a/envs/long_open_lock.py b/envs/long_open_lock.py
index 5aab093..df87742 100644
--- a/envs/long_open_lock.py
+++ b/envs/long_open_lock.py
@@ -9,12 +9,13 @@ from matplotlib import pyplot as plt
 from path import Path
 from sapienipc.ipc_utils.user_utils import ipc_update_render_all
 
-from envs.common_params import CommonParams
+
 
 script_path = os.path.dirname(os.path.realpath(__file__))
 repo_path = os.path.join(script_path, "..")
 sys.path.append(script_path)
 sys.path.append(repo_path)
+from envs.common_params import CommonParams
 import time
 from typing import List, Tuple
 
diff --git a/scripts/universal_training_script.py b/scripts/universal_training_script.py
index 1738c03..ef873c4 100644
--- a/scripts/universal_training_script.py
+++ b/scripts/universal_training_script.py
@@ -14,6 +14,7 @@ from path import Path
 from solutions.policies import (
     TD3PolicyForPointFlowEnv, TD3PolicyForLongOpenLockPointFlowEnv
 )
+
 from stable_baselines3 import TD3
 from stable_baselines3.common.callbacks import (CallbackList,
                                                 CheckpointCallback,
@@ -24,6 +25,7 @@ from utils.common import get_time
 from wandb.integration.sb3 import WandbCallback
 
 import wandb
+
 from arguments import *
 
 algorithm_aliases = {
diff --git a/solutions/networks.py b/solutions/networks.py
index 8e6f27e..3001c01 100644
--- a/solutions/networks.py
+++ b/solutions/networks.py
@@ -104,7 +104,7 @@ class PointNetFeatureExtractor(nn.Module):
         if marker_pos.ndim == 2:
             marker_pos = torch.unsqueeze(marker_pos, dim=0)
 
-        marker_pos = torch.transpose(marker_pos, 1, 2)
+        marker_pos = torch.transpose(marker_pos, 1, 2) # 感觉不需要 添加了变为(batch_num, self.pointnet_local_feature_num, point_num)
 
         batch_num = marker_pos.shape[0]
         point_num = marker_pos.shape[2]
